File: 202/py/202.py
class Solution:
    def isHappy(self, n: int) -> bool:
        seen = set()
        s = 0
        while n != 1 and n not in seen:
            seen.add(n)
            while n > 0 :
                num = n % 10
                s += num * num
                n = n // 10
            
            n = s
            s = 0
                    

        return n == 1

File: 202/go/202.go
func isHappy(n int) bool {
    num := 0
    sum := 0
    mapC := make(map[int]int)
    mapC[n] = 0

    for true {
        num = n % 10
        n = n / 10

        sum += num * num

        if n == 0 {
            if sum == 1 {
                return true
            }else if _, find := mapC[sum]; find{
                return false
            }else{
                n = sum
                sum = 0
                mapC[n] = 0
            }
        }
    }
    return false
}